### PR TITLE
see the content...

### PR DESCRIPTION
**Pull request by czhang03**
# Bug Fixes

- Fix a bug that when uses `Invoke-GitHubApi` with a header that already included `Authorization` that this will cause error

# Improvements / Enhancements

- Add `-preview` switch to `Invoke-GitHubApi` to enable Preview Api #29 
- Explicitly request v3 api, see: https://developer.github.com/v3/#current-version
- Add verbose output for request body, this makes debug a little easier
- Add `Get-GitHubLicense` function 

@pcgeek86

### CODE REVIEW

**Comment by czhang03** [comment]
hello?

File: Functions/Private/Invoke-GitHubApi.ps1 [hunk 1]
```diff
@@ -56,7 +81,10 @@
     Write-Verbose -Message ('Invoking the REST method: {0}' -f $ApiRequest.Uri)
         
     ### Append the HTTP message body (payload), if the caller specified one.
-    if ($Body) { $ApiRequest.Body = $Body; }
+    if ($Body) { 
+        $ApiRequest.Body = $Body 
+        Write-Verbose "the request body is {0}" -f $Body
```
Comment (pcgeek86) [inline Functions/Private/Invoke-GitHubApi.ps1]: Please use named parameters and parentheses as necessary.

File: Functions/Public/Get-GitHubLicense.ps1 [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    
+    process 
+    {
+        if (-Not ($LicenseId)) 
+        {
+            $restMethod = 'licenses'
+        }
+        else 
+        {
+            $restMethod = "licenses/{0}" -f $LicenseId
```
Comment (pcgeek86) [inline Functions/Public/Get-GitHubLicense.ps1]: Please use single quotes unless double quotes are absolutely necessary. In this case, they aren't.

File: Functions/Private/Invoke-GitHubApi.ps1 [hunk 3]
```diff
@@ -43,10 +48,30 @@
...
+    if ($Preview) {
+
+        if (!$Headers.Accept) {
+            $Headers.Add('Accept', 'application/vnd.github.drax-preview+json')
+        }
+        else {
+            $Headers.Accept = 'application/vnd.github.drax-preview+json'
+        }
+        
+        Write-Warning 'The API you are trying to retrive maybe preview'
```
Comment (pcgeek86) [inline Functions/Private/Invoke-GitHubApi.ps1]: Please use named parameters, not positional.

**Review by pcgeek86 — CHANGES_REQUESTED** [review]
Please review the comments and make changes as appropriate.

**Comment by czhang03** [comment]
a fix is pushed. Thank you.

**Comment by pcgeek86** [comment]
@chantisnake Thanks!

**Pull request merged**